examples/nature_of_code/chp_06_agents/6_01_seek.rs

```diff
@@ -0,0 +1,159 @@
+// The Nature of Code
+// Daniel Shiffman
+// http://natureofcode.com
+
+// Seeking "vehicle" follows the mouse position
+
+// Implements Craig Reynold's autonomous steering behaviors
+// One vehicle "seeks"
+// See: http://www.red3d.com/cwr/
+extern crate nannou;
+
+use nannou::prelude::*;
+use nannou::Draw;
+
+fn main() {
+    nannou::app(model, event, view).run();
+}
+
+struct Model {
+    vehicle: Vehicle,
+}
+
+struct Vehicle {
+    position: Vector2,
+    velocity: Vector2,
+    acceleration: Vector2,
+    r: f32,
+    // Maximum steering force
+    max_force: f32,
+    // Maximum speed
+    max_speed: f32,
+}
+
+impl Vehicle {
+    fn new(x: f32, y: f32) -> Self {
+        let position = Vector2::new(x, y);
+        let velocity = Vector2::new(0.0, -2.0);
+        let acceleration = Vector2::new(0.0, 0.0);
```

Comment:
In the prelude, nannou provides a short-hand for constructing a Vector2/3/4:

```rust
let acceleration = Vector2::new(0.0, 0.0); 
```
is equivalent to
```rust
let acceleration = vec2(0.0, 0.0);
```